test.py

```diff
@@ -1695,7 +1696,65 @@ def write_consolidated_boost_junit_xml(tmpdir: str, mode: str) -> None:
             test = ET.SubElement(suite, 'TestSuite', name=test_name)
         test.append(test_case)
     et = ET.ElementTree(xml)
-    et.write(f'{tmpdir}/{mode}/xml/boost.xunit.xml', encoding='unicode')
+    xunit_file = f'{tmpdir}/{mode}/xml/boost.xunit.xml'
+    et.write(xunit_file, encoding='unicode')
+    return xunit_file
+
+
+def boost_to_junit(boost_xml, junit_xml):
```

Comment:
Why not generate junit in the first place? https://www.boost.org/doc/libs/1_85_0/libs/test/doc/html/boost_test/test_output/log_formats/log_junit_format.html

We can generate only junit, if Jenkins understands it.

Reply:
It was the first approach that I've tried. But unfortunately, several tests start to fail when I'm adding the logger. The first two runs faced this issue. Locally I was able to fix two of the tests, just increased memory from 2G to 3G. The last one [dirty_memory_manager_test](https://github.com/scylladb/scylladb/blob/master/test/boost/dirty_memory_manager_test.cc) was failing even when I was increasing the memory to 20G. Today, I've found out how to mitigate this. But not sure if this is the best solution. More about it [here](https://github.com/scylladb/scylladb/pull/19925#discussion_r1711817036)